### ISSUE TITLE
[7.2.x] Upgrade to Jetty 9.4.48.v20220622

### ISSUE THREAD

**Issue by grgrzybek** [ops4j/org.ops4j.pax.web#1748]
Upgrade for 7.3: #1742

**Comment by grgrzybek** [comment]
Non-Karaf tests pass. Checking Karaf test failures.

**Comment by grgrzybek** [comment]
`org.ops4j.pax.web.itest.karaf.SpringOsgiKarafTest.testWC()`:
```
Caused by: java.lang.ClassNotFoundException: org.apache.geronimo.components.jaspi.AuthConfigFactoryImpl not found by org.apache.geronimo.specs.geronimo-jaspic_1.0_spec [38]
	at org.apache.felix.framework.BundleWiringImpl.findClassOrResourceByDelegation(BundleWiringImpl.java:1639) ~[?:?]
	at org.apache.felix.framework.BundleWiringImpl.access$200(BundleWiringImpl.java:80) ~[?:?]
	at org.apache.felix.framework.BundleWiringImpl$BundleClassLoader.loadClass(BundleWiringImpl.java:2053) ~[?:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:351) ~[?:1.8.0_342]
	at java.lang.Class.forName0(Native Method) ~[?:1.8.0_342]
	at java.lang.Class.forName(Class.java:348) ~[?:1.8.0_342]
	at org.apache.geronimo.osgi.locator.ProviderLocator.loadClass(ProviderLocator.java:195) ~[?:?]
	at javax.security.auth.message.config.AuthConfigFactory$3.run(AuthConfigFactory.java:68) ~[?:?]
	at java.security.AccessController.doPrivileged(Native Method) ~[?:1.8.0_342]
	at javax.security.auth.message.config.AuthConfigFactory.getFactory(AuthConfigFactory.java:64) ~[?:?]
	... 47 more
```

**Comment by grgrzybek** [comment]
@jbonofre ping

for long time, `pax-http-tomcat` and `pax-http-undertow` features in Pax Web 7.2.x/7.3.x had:
```xml
    <feature name="pax-http-undertow" version="${project.version}">
...
        <feature version="[7.2,7.3)">pax-http</feature>
...
        <capability>
            pax.http.provider;provider:=undertow,
            osgi.service;effective:=active;objectClass=org.xnio.XnioProvider
        </capability>
```

And `pax-http` feature was:
```xml
<feature name="pax-http" version="${project.version}" description="Implementation of the OSGI HTTP Service">
    <details>Allows to publish servlets using pax web and jetty</details>
    <feature dependency="true" version="[7.2,7.3)">pax-http-jetty</feature>
    <requirement>
        pax.http.provider
    </requirement>
</feature>
```

Which contains `dependency="true"` on `pax-http-jetty.

When running Pax Web Karaf itests with Karaf 4.2.11 I see that simply installing `pax-http-undertow` feature leads to installation of `pax-http-jetty` features - despite the `dependency="true"` flag.

Is it something related to Karaf resolver? It's like `pax-http-undertow` feature doesn't see its own capability named `pax.http.provider`...

**Comment by grgrzybek** [comment]
The problem I see is that `etc/org.ops4j.pax.web.cfg` is added by Jetty version and it has:
```
org.ops4j.pax.web.config.file=${karaf.etc}/jetty.xml
```

So starting Undertow server leads to:
```
Caused by: javax.xml.bind.UnmarshalException: unexpected element (uri:"", local:"Configure"). Expected elements are <{urn:org.ops4j.pax.web:undertow:1.0}undertow>
	at com.sun.xml.internal.bind.v2.runtime.unmarshaller.UnmarshallingContext.handleEvent(UnmarshallingContext.java:726) ~[?:1.8.0_342]
	at com.sun.xml.internal.bind.v2.runtime.unmarshaller.Loader.reportError(Loader.java:247) ~[?:1.8.0_342]
	at com.sun.xml.internal.bind.v2.runtime.unmarshaller.Loader.reportError(Loader.java:242) ~[?:1.8.0_342]
	at com.sun.xml.internal.bind.v2.runtime.unmarshaller.Loader.reportUnexpectedChildElement(Loader.java:109) ~[?:1.8.0_342]
	at com.sun.xml.internal.bind.v2.runtime.unmarshaller.UnmarshallingContext$DefaultRootLoader.childElement(UnmarshallingContext.java:1131) ~[?:1.8.0_342]
	at com.sun.xml.internal.bind.v2.runtime.unmarshaller.UnmarshallingContext._startElement(UnmarshallingContext.java:556) ~[?:1.8.0_342]
	at com.sun.xml.internal.bind.v2.runtime.unmarshaller.UnmarshallingContext.startElement(UnmarshallingContext.java:538) ~[?:1.8.0_342]
	at com.sun.xml.internal.bind.v2.runtime.unmarshaller.InterningXmlVisitor.startElement(InterningXmlVisitor.java:60) ~[?:1.8.0_342]
	at com.sun.xml.internal.bind.v2.runtime.unmarshaller.SAXConnector.startElement(SAXConnector.java:153) ~[?:1.8.0_342]
	at org.ops4j.pax.web.service.undertow.internal.configuration.ResolvingContentHandler.startElement(ResolvingContentHandler.java:73) ~[?:?]
...
```

**Comment by grgrzybek** [comment]
Problem fixed by reverting (in 7.2 only) 7128d75c5fe2e8e91f242a7352d0a849fd0e903b - checking the impact.

**Comment by grgrzybek** [comment]
After removing `<feature version="[7.2,7.3)">pax-http</feature>` from `pax-http-undertow`, the feature installs fine, but installation of `pax-war`, which uses `pax-http-whiteboard` still installs Jetty, because `pax-http-whiteboard` uses this `<feature version="[7.2,7.3)">pax-http</feature>` dependency...

I'm pretty sure it wasn't a problem some time ago...

**Comment by grgrzybek** [comment]
Hmm, I see the same failure with tag `7.2.29`...

**Comment by grgrzybek** [comment]
worked fine with tag `7.2.18` and Karaf `4.2.1`...

**Comment by grgrzybek** [comment]
With Karaf `4.2.17-SNAPSHOT` using Pax Web `7.2.30-SNAPSHOT` I have only one failure:
```
[ERROR] Failures: 
[ERROR]   WebSocketExampleTest.test:73
[INFO] 
[ERROR] Tests run: 318, Failures: 1, Errors: 0, Skipped: 5
```

Probably because I've reverted 7128d75c5fe2e8e91f242a7352d0a849fd0e903b.

**Comment by grgrzybek** [comment]
@jbonofre works well when Karaf 4.2.17-SNAPSHOT has this:
```
$ git diff 
diff --git a/assemblies/features/standard/src/main/feature/feature.xml b/assemblies/features/standard/src/main/feature/feature.xml
index e7ea86c4557..5e7b5084803 100644
--- a/assemblies/features/standard/src/main/feature/feature.xml
+++ b/assemblies/features/standard/src/main/feature/feature.xml
@@ -892,7 +892,6 @@ org.osgi.service.http.port=${env:ORG_APACHE_FELIX_HTTP_ORG_OSGI_SERVICE_HTTP_POR
         <bundle dependency="true" start-level="30">mvn:javax.mail/mail/${javax.mail.version}</bundle>
         <bundle dependency="true" start-level="30">mvn:org.apache.geronimo.specs/geronimo-jta_1.1_spec/${geronimo.jta-spec.version}</bundle>
         <bundle dependency="true" start-level="30">mvn:javax.annotation/javax.annotation-api/${javax.annotation.version}</bundle>
-        <bundle dependency="true" start-level="30">mvn:org.apache.geronimo.specs/geronimo-jaspic_1.0_spec/${geronimo.jaspic-spec.version}</bundle>
         <bundle dependency="true" start-level="30">mvn:org.ow2.asm/asm/${asm.version}</bundle>
         <bundle dependency="true" start-level="30">mvn:org.ow2.asm/asm-util/${asm.version}</bundle>
         <bundle dependency="true" start-level="30">mvn:org.ow2.asm/asm-commons/${asm.version}</bundle>
@@ -902,7 +901,6 @@ org.osgi.service.http.port=${env:ORG_APACHE_FELIX_HTTP_ORG_OSGI_SERVICE_HTTP_POR
         <bundle start-level="30">mvn:org.eclipse.jetty/jetty-continuation/${jetty.version}</bundle>
         <bundle start-level="30">mvn:org.eclipse.jetty/jetty-http/${jetty.version}</bundle>
         <bundle start-level="30">mvn:org.eclipse.jetty/jetty-io/${jetty.version}</bundle>
-        <bundle start-level="30">mvn:org.eclipse.jetty/jetty-jaspi/${jetty.version}</bundle>
         <bundle start-level="30">mvn:org.eclipse.jetty/jetty-plus/${jetty.version}</bundle>
                <bundle start-level="30">mvn:org.eclipse.jetty/jetty-jndi/${jetty.version}</bundle>
         <bundle start-level="30">mvn:org.eclipse.jetty/jetty-rewrite/${jetty.version}</bundle>
diff --git a/pom.xml b/pom.xml
index 3d8fa1403a4..0a7829a3370 100644
--- a/pom.xml
+++ b/pom.xml
@@ -290,7 +290,7 @@
         <httpclient.version>4.5.13</httpclient.version>
         <jansi.version>1.18</jansi.version>
         <javassist.version>3.9.0.GA</javassist.version>
-        <jetty.version>9.4.43.v20210629</jetty.version>
+        <jetty.version>9.4.48.v20220622</jetty.version>
         <jline.version>3.19.0</jline.version>
         <junit.version>4.13.2</junit.version>
         <jsw.version>3.2.3</jsw.version>
@@ -310,7 +310,7 @@
         <pax.base.version>1.5.1</pax.base.version>
         <pax.swissbox.version>1.8.5</pax.swissbox.version>
         <pax.url.version>2.6.10</pax.url.version>
-        <pax.web.version>7.2.29</pax.web.version>
+        <pax.web.version>7.2.30-SNAPSHOT</pax.web.version>
         <pax.tinybundle.version>3.0.0</pax.tinybundle.version>
         <pax.jdbc.version>1.5.0</pax.jdbc.version>
         <pax.jms.version>1.1.0</pax.jms.version>
```

**Comment by grgrzybek** [comment]
Mind that JASPIC isn't supported by Pax Web anyway

**Comment by grgrzybek** [comment]
7.2.30 is ready to be released. `WarUndertowKarafTest` didn't work for some time. But all other tests work.
As it's maintenance release, I think it's fine to release with Jetty 9.4.48 upgrade.

**Comment by dstoch** [comment]
Thanks for this upgrade. Now we have to wait for the release ;).

**Comment by grgrzybek** [comment]
I've released 8.0.7 today, so I can release 7.2.x tomorrow ;).
